File: src/main/java/io/gittul/domain/tag/TagRepository.java
package io.gittul.domain.tag;

import io.gittul.domain.tag.entity.Tag;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Set;

@Repository
public interface TagRepository extends JpaRepository<Tag, Long> {
    Set<Tag> findAllByTagNameIn(List<String> tagNames);
}

File: app/src/main/kotlin/io/gittul/app/domain/tag/TagRepository.kt
package io.gittul.app.domain.tag

import io.gittul.core.domain.tag.entity.Tag
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TagRepository : JpaRepository<Tag, Long> {
    fun findAllByTagNameIn(tagNames: MutableList<String>): MutableSet<Tag>
}
